File: suc/swift/016.swift
import SwiftUI

// MARK: - AppTokens
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0xFF00FFFF)
        static let secondary = Color(hex: 0xFF9D00FF)
        static let tertiary = Color(hex: 0xFF00FFA3)
        static let background = Color(hex: 0xFF0B0B0D)
        static let surface = Color(hex: 0xFF1A1A1D)
        static let surfaceVariant = Color(hex: 0xFF2E2E33)
        static let outline = Color(hex: 0xFF3F3F46)
        static let success = Color(hex: 0xFF22C55E)
        static let warning = Color(hex: 0xFFF59E0B)
        static let error = Color(hex: 0xFFEF4444)
        static let onPrimary = Color(hex: 0xFF000000)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0xFF000000)
        static let onBackground = Color(hex: 0xFFE5E7EB)
        static let onSurface = Color(hex: 0xFFE5E7EB)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 28, weight: .bold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        static let smallCornerRadius: CGFloat = 6.0
        static let mediumCornerRadius: CGFloat = 10.0
        static let largeCornerRadius: CGFloat = 14.0
    }

    struct Spacing {
        static let xs: CGFloat = 4.0
        static let sm: CGFloat = 8.0
        static let md: CGFloat = 12.0
        static let lg: CGFloat = 16.0
        static let xl: CGFloat = 24.0
        static let xxl: CGFloat = 32.0
    }
}

// MARK: - Color Extension for Hex Initialization
extension Color {
    init(hex: UInt) {
        self.init(
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0,
            opacity: Double((hex >> 24) & 0xFF) / 255.0 == 0 && hex <= 0xFFFFFF ? 1 : Double((hex >> 24) & 0xFF) / 255.0
        )
    }
}

// MARK: - RootScreen
struct ContentView: View {
    @State private var title: String = ""
    @State private var recording: Bool = false
    @State private var seconds: Int = 0
    @State private var timer: Timer?

    var body: some View {
        VStack(spacing: 0) {
            // 顶部栏
            Text("Neon Recorder")
                .font(AppTokens.TypographyTokens.display)
                .foregroundColor(AppTokens.Colors.primary)
                .padding(.vertical, AppTokens.Spacing.lg)
                .frame(maxWidth: .infinity)
                .background(AppTokens.Colors.background)
            
            // 内容区域
            VStack(spacing: AppTokens.Spacing.lg) {
                // 录音名称输入框
                VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
                    Text("Recording name")
                        .font(AppTokens.TypographyTokens.label)
                        .foregroundColor(AppTokens.Colors.onSurface.opacity(0.7))
                    
                    TextField("", text: $title)
                        .font(AppTokens.TypographyTokens.body)
                        .foregroundColor(AppTokens.Colors.onSurface)
                        .padding(AppTokens.Spacing.md)
                        .background(AppTokens.Colors.surface)
                        .cornerRadius(AppTokens.Shapes.mediumCornerRadius)
                        .overlay(
                            RoundedRectangle(cornerRadius: AppTokens.Shapes.mediumCornerRadius)
                                .stroke(title.isEmpty ? AppTokens.Colors.outline : AppTokens.Colors.primary, lineWidth: 1)
                        )
                        .accentColor(AppTokens.Colors.primary)
                }
                
                // 录音状态卡片
                ZStack {
                    RoundedRectangle(cornerRadius: AppTokens.Shapes.largeCornerRadius)
                        .fill(AppTokens.Colors.surfaceVariant)
                    
                    Text(recording ? "Recording... \(seconds)s" : "Press Record")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(recording ? AppTokens.Colors.primary : AppTokens.Colors.onSurface)
                }
                .frame(height: 200)
                
                // 按钮行
                HStack(spacing: AppTokens.Spacing.xl) {
                    Button(action: {
                        recording.toggle()
                        if recording {
                            seconds = 0
                            startTimer()
                        } else {
                            stopTimer()
                        }
                    }) {
                        Text(recording ? "Stop" : "Record")
                            .font(AppTokens.TypographyTokens.title)
                            .foregroundColor(AppTokens.Colors.onPrimary)
                            .frame(width: 160, height: 52)
                            .background(
                                RoundedRectangle(cornerRadius: AppTokens.Shapes.largeCornerRadius)
                                    .fill(recording ? AppTokens.Colors.error : AppTokens.Colors.primary)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())

                    Button(action: {
                        seconds = 0
                        if recording {
                            stopTimer()
                            startTimer()
                        }
                    }) {
                        Text("Reset")
                            .font(AppTokens.TypographyTokens.title)
                            .foregroundColor(AppTokens.Colors.onSecondary)
                            .frame(width: 120, height: 52)
                            .background(
                                RoundedRectangle(cornerRadius: AppTokens.Shapes.largeCornerRadius)
                                    .fill(AppTokens.Colors.secondary)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())
                }
                
                Spacer()
                    .frame(height: AppTokens.Spacing.xxl)
                
                Text("Output Path: /storage/recordings/\(title.isEmpty ? "untitled" : title).mp3")
                    .font(AppTokens.TypographyTokens.body)
                    .foregroundColor(AppTokens.Colors.onSurface.opacity(0.7))
                    .frame(maxWidth: .infinity, alignment: .leading)
            }
            .padding(AppTokens.Spacing.lg)
            .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .top)
            .background(
                LinearGradient(
                    gradient: Gradient(colors: [AppTokens.Colors.background, AppTokens.Colors.surfaceVariant]),
                    startPoint: .top,
                    endPoint: .bottom
                )
            )
        }
        .background(AppTokens.Colors.background)
        .ignoresSafeArea(.keyboard, edges: .bottom)
        .ignoresSafeArea(.all, edges: .all)
        .statusBar(hidden: true)
    }
    
    // MARK: - Timer Functions
    private func startTimer() {
        stopTimer()
        timer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true) { _ in
            seconds += 1
        }
    }
    
    private func stopTimer() {
        timer?.invalidate()
        timer = nil
    }
}

// MARK: - App Entry Point
@main
struct NeonRecorderApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
                .preferredColorScheme(.dark)
        }
    }
}

File: suc/kotlin/016.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "015Recorderen"
private const val UI_TYPE = "Media"
private const val STYLE_THEME = "Dark Neon"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF00FFFF)
        val secondary = Color(0xFF9D00FF)
        val tertiary = Color(0xFF00FFA3)
        val background = Color(0xFF0B0B0D)
        val surface = Color(0xFF1A1A1D)
        val surfaceVariant = Color(0xFF2E2E33)
        val outline = Color(0xFF3F3F46)
        val success = Color(0xFF22C55E)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFF000000)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFF000000)
        val onBackground = Color(0xFFE5E7EB)
        val onSurface = Color(0xFFE5E7EB)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(10.dp)
        val large = RoundedCornerShape(14.dp)
    }
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
        val xxl = 32.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f)
        val level2 = ShadowSpec(4.dp, 8.dp, 4.dp, 0.16f)
    }
}

private val AppColorScheme = darkColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    var title by remember { mutableStateOf("") }
    var recording by remember { mutableStateOf(false) }
    var seconds by remember { mutableIntStateOf(0) }

    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Text("Neon Recorder", style = MaterialTheme.typography.displayLarge, color = AppTokens.Colors.primary)
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = AppTokens.Colors.background)
            )
        },
        containerColor = AppTokens.Colors.background
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .background(
                    Brush.verticalGradient(
                        listOf(AppTokens.Colors.background, AppTokens.Colors.surfaceVariant)
                    )
                )
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            OutlinedTextField(
                value = title,
                onValueChange = { title = it },
                label = { Text("Recording name", color = AppTokens.Colors.onSurface) },
                singleLine = true,
                shape = AppTokens.Shapes.medium,
                colors = OutlinedTextFieldDefaults.colors(
                    focusedBorderColor = AppTokens.Colors.primary,
                    unfocusedBorderColor = AppTokens.Colors.outline,
                    cursorColor = AppTokens.Colors.primary,
                    focusedTextColor = AppTokens.Colors.onSurface,
                    unfocusedTextColor = AppTokens.Colors.onSurface
                ),
                modifier = Modifier.fillMaxWidth()
            )
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp),
                shape = AppTokens.Shapes.large,
                colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surfaceVariant)
            ) {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = if (recording) "Recording... ${seconds}s" else "Press Record",
                        style = MaterialTheme.typography.titleMedium,
                        color = if (recording) AppTokens.Colors.primary else AppTokens.Colors.onSurface
                    )
                }
            }
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceEvenly
            ) {
                Button(
                    onClick = {
                        recording = !recording
                        if (recording) seconds = 0
                    },
                    shape = AppTokens.Shapes.large,
                    colors = ButtonDefaults.buttonColors(
                        containerColor = if (recording) AppTokens.Colors.error else AppTokens.Colors.primary,
                        contentColor = AppTokens.Colors.onPrimary
                    ),
                    modifier = Modifier.height(52.dp).width(160.dp)
                ) {
                    Text(if (recording) "Stop" else "Record", style = MaterialTheme.typography.titleMedium)
                }
                Button(
                    onClick = { seconds = 0 },
                    shape = AppTokens.Shapes.large,
                    colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.secondary, contentColor = AppTokens.Colors.onSecondary),
                    modifier = Modifier.height(52.dp).width(120.dp)
                ) {
                    Text("Reset", style = MaterialTheme.typography.titleMedium)
                }
            }
            Spacer(modifier = Modifier.height(AppTokens.Spacing.xxl))
            Text(
                text = "Output Path: /storage/recordings/${if (title.isBlank()) "untitled" else title}.mp3",
                style = MaterialTheme.typography.bodyMedium,
                color = AppTokens.Colors.onSurface.copy(alpha = 0.7f)
            )
        }
    }


}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFF0B0B0D)
@Composable
fun PreviewScreen() {
    AppTheme {
        RootScreen()
    }
}